ddtrace/internal/datadog/profiling/dd_wrapper/src/sample.cpp

```diff
@@ -0,0 +1,323 @@
+    const std::string_view key_sv = to_string(key);
+    if (key_sv.empty()) {
+        return true;
+    }
+
+    labels[cur_label].key = to_slice(key_sv);
+    labels[cur_label].str = to_slice("");
+    labels[cur_label].num = val;
+    labels[cur_label].num_unit = to_slice("");
+    cur_label++;
+    return true;
+}
+
+void
+Sample::clear_buffers()
+{
+    std::fill(values.begin(), values.end(), 0);
+    std::fill(std::begin(labels), std::end(labels), ddog_prof_Label{});
+    locations.clear();
+    cur_label = 0;
+    dropped_frames = 0;
+}
+
+bool
+Sample::flush_sample()
+{
+    if (dropped_frames > 0) {
+        const std::string name =
+          "<" + std::to_string(dropped_frames) + " frame" + (1 == dropped_frames ? "" : "s") + " omitted>";
+        Sample::push_frame_impl(name, "", 0, 0);
+    }
+
+    const ddog_prof_Sample sample = {
+        .locations = { locations.data(), locations.size() },
+        .values = { values.data(), values.size() },
+        .labels = { labels.data(), cur_label },
+    };
+
+    const bool ret = profile_state.collect(sample);
+    clear_buffers();
+    return ret;
+}
+
+bool
+Sample::push_cputime(int64_t cputime, int64_t count)
+{
+    // NB all push-type operations return bool for semantic uniformity,
+    // even if they can't error.  This should promote generic code.
+    if (type_mask & SampleType::CPU) {
+        values[profile_state.val().cpu_time] += cputime * count;
+        values[profile_state.val().cpu_count] += count;
+        return true;
+    }
+    std::cout << "bad push cpu" << std::endl;
+    return false;
+}
+
+bool
+Sample::push_walltime(int64_t walltime, int64_t count)
+{
+    if (type_mask & SampleType::Wall) {
+        values[profile_state.val().wall_time] += walltime * count;
+        values[profile_state.val().wall_count] += count;
+        return true;
+    }
+    std::cout << "bad push wall" << std::endl;
+    return false;
+}
+
+bool
+Sample::push_exceptioninfo(std::string_view exception_type, int64_t count)
+{
+    if (type_mask & SampleType::Exception) {
+        push_label(ExportLabelKey::exception_type, exception_type);
+        values[profile_state.val().exception_count] += count;
+        return true;
+    }
+    std::cout << "bad push except" << std::endl;
+    return false;
+}
+
+bool
+Sample::push_acquire(int64_t acquire_time, int64_t count)
+{
+    if (type_mask & SampleType::LockAcquire) {
+        values[profile_state.val().lock_acquire_time] += acquire_time;
+        values[profile_state.val().lock_acquire_count] += count;
+        return true;
+    }
+    std::cout << "bad push acquire" << std::endl;
+    return false;
+}
+
+bool
+Sample::push_release(int64_t lock_time, int64_t count)
+{
+    if (type_mask & SampleType::LockRelease) {
+        values[profile_state.val().lock_release_time] += lock_time;
+        values[profile_state.val().lock_release_count] += count;
+        return true;
+    }
+    std::cout << "bad push release" << std::endl;
```

Comment:
I know this is a probably a temporary workaround for logging. 
Perhaps we can just plug a log_once to stderr before we figure out how to do the logger compatible with python ?

Reply:
I have a short-term todo to just plug into the dd-trace-py logger.  I was going to do it in this PR, but it got way too big.